### PR TITLE
Privacy improvements & other minor changes

### PR DESCRIPTION
**Pull request by conorwastakenwastaken**
# Private Threads

Threads can now be private. Private threads can only be viewed by anyone who has a role which is inside of the `inboxAdminRoleIDs` array. Admins can make a thread private with the new `!hide` command, and threads are automatically made private when moved to a category other than the `newThreadCategoryId`.

 - `isPrivate` data migration (977e10d)
 - Added `!hide` command and functions to make threads private/public (7e6c96b)
 - Added oAuth2 function to check if someone is a dash admin, also redone how users are authorized into the dash so that the web server can get the authorized user as well (c6a2ee9)
 - Don't allow non-Admins to view private or open threads (0c6d99d)
 - Don't list private threads to non-Admins inside of `!logs` (565ad7b)

# New Config Property - `adminMentionRole`

 - This property controls whether admins should be mentioned when someone moves a thread to any category other than the `newThreadCategoryId`
 - If the user who moves the thread has a role inside of the `inboxAdminRoleIDs` array, it won't mention the `adminMentionRole`

# Misc Changes

 - Change how admin roles work, you can now have more than 1 (aced3be)
 - The `!purge` command requires an argument (and said argument has to be a number), it will no longer purge up to 100 messages if you run it without any arguments (98cd283)
 - Fixed an issue where the `!logs` command wouldn't respond if ran outside of a thread with no arguments (0fed4ff)
 - Made `config2.json` neater by putting the properties in the same order as the `config.js` file, as well as removing options which are already set by default in `config.js` (36c1f42)

### CODE REVIEW

File: src/oauth2.js [hunk 1]
```diff
@@ -35,33 +35,57 @@ function login (req, res) {
...
-    let token = req.cookies.token;
-    try {
-      let accessToken = jwt.verify(token, config.clientSecret).token;
-      let response = await superagent.get("https://discord.com/api/users/@me")
-      .set("Authorization", `Bearer ${accessToken}`);
-      req.user = response.body;
-    } catch (err) {
-      return res.redirect("/login");
-    }
+    req.user = await getAuthUser(res.cookies.token);
```
Comment (beanjo55) [inline src/oauth2.js]: should this be req instead of res, for the cookie?

File: src/modules/webserver.js [hunk 2]
```diff
@@ -55,6 +51,14 @@ function getAttachment (id, desiredFilename) {
   return [mime.getType(ext), fs.readFileSync(attachmentPath)];
 }
 
+/**
+ * @param {express.Response} res
+ */
+async function viewPrivateThreads(res) {
+  const user = await oauth2.getAuthUser(res.cookies.token);
```
Comment (beanjo55) [inline src/modules/webserver.js]: same here, res instead of req

**Review by beanjo55 — CHANGES_REQUESTED** [review]
looks good, just need the cookie accesses changed

File: src/oauth2.js (hunk 1 above)
Comment (conorwastakenwastaken) [reply src/oauth2.js]: Whoops lol, tysm

**Pull request merged**